src/librustc_metadata/decoder.rs

```diff
@@ -357,17 +369,18 @@ impl<'a, 'tcx> SpecializedDecoder<ty::GenericPredicates<'tcx>> for DecodeContext
         Ok(ty::GenericPredicates {
             parent: Decodable::decode(self)?,
             predicates: (0..self.read_usize()?).map(|_| {
-                // Handle shorthands first, if we have an usize > 0x80.
-                if self.opaque.data[self.opaque.position()] & 0x80 != 0 {
-                    let pos = self.read_usize()?;
-                    assert!(pos >= SHORTHAND_OFFSET);
-                    let pos = pos - SHORTHAND_OFFSET;
-
-                    self.with_position(pos, ty::Predicate::decode)
-                } else {
-                    ty::Predicate::decode(self)
-                }
-            }).collect::<Result<Vec<_>, _>>()?
+                    // Handle shorthands first, if we have an usize > 0x80.
```

Comment:
This indentation is wrong (indented 8 spaces further than the previous line) -- looks like a `rustfmt` bug.
The indentation would make more sense if `.map(|_| {` had its own line (but I still prefer the original formatting).